### PR TITLE
Support token history

### LINKED ISSUE #6111: Token history table
### Problem

As a user, I'd like to capture all changes to my token type.

### Solution

See https://github.com/hashgraph/hedera-mirror-node/pull/4557/files for inspiration

* Change `token.modified_timestamp` to be a `timestamp_range` int8range
* Add a `token_history` table with the same fields as `token`
* Add the same primary key but include timestamp_range
* Add a gist timestamp_range on history
* Remove `TokenId` class in favor of directly embedding `EntityId tokenId`
* Move `Token` domain to `AbstractToken`
* Add a `Token` and `TokenHistory` classes that extends `AbstractToken`
* Add `DomainBuilder.tokenHistory()`
* Add `TokenHistoryRepository`
* Update transaction handlers
* Update `SqlEntityListener.onToken()` to merge changes from other tokens by id
* Update any APIs that reference `token.modified_timestamp`

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Description**:

* Add `List`, `Range` and `EntityId` types to global setup to avoid per domain annotations
* Add a `skipPartialUpdate` property to `@Upsertable` to skip updates without an existing row
* Add a `token_history` table
* Change `/api/v1/tokens` to use `timestamp_range`
* Change `token.modified_timestamp` to `token.timestamp_range`
* Fix flaky `BatchUpserterTest.tokenDissociateTransfer()`
* Fix not including `nft_history` in pg_partman maintenance
* Fix not merging `token.fee_schedule_key` properly
* Remove `TokenId` class in favor of `Long`

**Related issue(s)**:

Fixes #6111

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by xin-hedera — CHANGES_REQUESTED** [review]

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/sql/SqlEntityListener.java [hunk 1]
```diff
@@ -406,8 +408,10 @@ public void onStakingRewardTransfer(StakingRewardTransfer stakingRewardTransfer)
 
     @Override
     public void onToken(Token token) throws ImporterException {
-        // tokens could experience multiple updates in a single record file, handle updates in memory for this case
-        tokens.merge(token.getTokenId().getTokenId().getId(), token, this::mergeToken);
+        var merged = tokenState.merge(token.getTokenId(), token, this::mergeToken);
+        if (merged == token) {
+            tokens.add(token);
+        }
```
[the remark this answers is not in the document]
Comment (steven-sheehy) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/sql/SqlEntityListener.java]: Done

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/sql/SqlEntityListener.java [hunk 2]
```diff
@@ -574,7 +579,7 @@ private void flush() {
     private void flushNftState() {
         try {
             // flush tables required for an accurate nft state in database to ensure correct state-dependent changes
-            batchPersister.persist(tokens.values());
+            batchPersister.persist(tokens);
```
[the remark this answers is not in the document]
Comment (steven-sheehy) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/sql/SqlEntityListener.java:580]: Done

File: hedera-mirror-common/src/main/java/com/hedera/mirror/common/domain/package-info.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+@ConverterRegistration(converter = UnknownIdConverter.class)
```
[the remark this answers is not in the document]
Comment (steven-sheehy) [reply hedera-mirror-common/src/main/java/com/hedera/mirror/common/domain/package-info.java:17]: Done

File: hedera-mirror-common/src/main/java/com/hedera/mirror/common/domain/token/AbstractTokenAccount.java [hunk 4]
```diff
@@ -51,6 +52,7 @@ public class AbstractTokenAccount implements History {
 
     private Boolean automaticAssociation;
 
+    @Column(updatable = false)
```
[the remark this answers is not in the document]
Comment (steven-sheehy) [reply hedera-mirror-common/src/main/java/com/hedera/mirror/common/domain/token/AbstractTokenAccount.java]: Done

**Review by xin-hedera — APPROVED** [review]
LGTM

File: hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/batch/BatchInserterTest.java [hunk 5]
```diff
@@ -42,55 +39,41 @@
...
-    @Resource
-    private TokenTransferRepository tokenTransferRepository;
-
-    @Resource
-    private TransactionRepository transactionRepository;
+    private final BatchPersister batchInserter;
+    private final CryptoTransferRepository cryptoTransferRepository;
+    private final TopicMessageRepository topicMessageRepository;
+    private final TokenTransferRepository tokenTransferRepository;
+    private final TransactionRepository transactionRepository;
```
[the remark this answers is not in the document]
Comment (steven-sheehy) [reply hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/batch/BatchInserterTest.java]: Removed

**Review by jnels124 — APPROVED** [review]

**Pull request merged**